### PR TITLE
reverse calibration lut && include kersten recalibration

### PR DESCRIPTION
**Pull request by vincelhx**

### CODE REVIEW

**Review by agrouaze — COMMENTED** [review]
What happend if I apply this function to a SLC product?
reading the code, it seems to only print a warning. Why not a raise Error ?

**Comment by vincelhx** [comment]
yes at this point it shoud raise an Error

**Comment by vincelhx** [comment]
Hi. Now it raises an Error

**Comment by agrouaze** [comment]
@vincelhx Is possible to add the `xsar.open_dataset` times from `develop` branch and the ones from your branch with and without recalibration applied?
Is your re-calibration working for WV, SM ,EW too ?

**Comment by vincelhx** [comment]
its working for GRD only : i add an exception?
what do you mean by ```xsar.open_dataset``̀̀` times?

**Comment by agrouaze** [comment]
> its working for GRD only : i add an exception? 
WV cannot be produced as GRD but SM and EW are available in GRD, so I guess that re-calibration should work the same way for all acquisition modes. Correct? It is not the case or if it is not yet implemented we need to clarify it in the code and in the doc.
> what do you mean by `xsar.open_dataset`̀ times?
I simply mean the time that opening a product takes, to comapre before/after with/without your re-calibration.

**Comment by vincelhx** [comment]
to open a new recalibrated dataset : 
`s1ds_recal = xsar.Sentinel1Dataset(L1_path,resolution='1000m',recalibration=True)`

if no roll angle ( no recalibration ) => `ValueError: Recalibration can't be done without roll angle. You probably work with an old file for which roll angle is not auxiliary file.`

only for EW/IW

**Comment by vincelhx** [comment]
var concerned : `INTEREST_VAR = ["sigma0_raw","gamma0_raw","beta0_raw"]` (l660 from sentinel1_dataset.py)

**Comment by vincelhx** [comment]
no change in .dataset : 
excepted an attribute to comment if recalib was applied or not 
image

**Comment by vincelhx** [comment]
new datatree related to recalibration steps :
maybe it need to be completed or detailed. 

image

**Comment by vincelhx** [comment]
mean times to  ```xsar.Sentinel1Dataset(L1_path, resolution = '1000m')```

before commit without recalibration :
Mean time for S1A_IW_GRDH_1SDV_20141231T150741_20141231T150810_003965_004C59_FA9B : 4.936 secondes
Mean time for S1B_EW_GRDM_1SDV_20190306T135326_20190306T135430_015234_01C808_DE3D : 4.156 secondes

after commit without recalibration : 
Mean time for S1A_IW_GRDH_1SDV_20141231T150741_20141231T150810_003965_004C59_FA9B : 4.525 secondes
Mean time for S1B_EW_GRDM_1SDV_20190306T135326_20190306T135430_015234_01C808_DE3D : 6.172 secondes
this is due to changes in xarray-safe-s1 https://github.com/umr-lops/xarray-safe-s1/pull/15

after commit with recalibraton 
S1A_IW_GRDH_1SDV_20141231T150741_20141231T150810_003965_004C59_FA9B : error (old file)
Mean time for S1B_EW_GRDM_1SDV_20190306T135326_20190306T135430_015234_01C808_DE3D : 6.067 secondes

File: src/xsar/sentinel1_dataset.py [hunk 1]
```diff
@@ -377,6 +449,10 @@ def add_high_resolution_variables(self, luts=False, patch_variable=True, skip_va
 
             self._dataset = self._dataset.merge(self._load_from_geoloc(geoloc_vars, lazy_loading=lazy_loading))
 
+            if self.apply_recalibration:
+                self.add_swath_number()
+                self.add_gains('S1B_AUX_CAL_V20160422T000000_G20210104T140113.SAFE','S1B_AUX_PP1_V20160422T000000_G20220323T140710.SAFE')
+
```
Comment (vincelhx) [inline src/xsar/sentinel1_dataset.py:455]: commentaire

**Review by agrouaze — APPROVED** [review]

**Pull request merged**